python/tests/test_aggregates.py

```diff
@@ -0,0 +1,335 @@
+        
+        sql.start()
+
+        sql.input_json("avg0_value_groupby", data)
+        
+        # checked manually
+        expected_data = [
+            {"id": 0, "c1": 1, "c2": 11, "c3": 16, "c4": 22, "c5": 5, "c6": 33, "c7": 70, "c8": 44},
+            {"id" : 1, "c1": 11, "c2": 22, "c3" : None, "c4" : 45, "c5" : 51, "c6" : 61, "c7":72, "c8": 88}]
+        
+        sql.wait_for_completion(True)
+        
+        out_data = out.to_dict()
+        print(type(out_data))  
+        
+        def check_dict_types(out_data):
+            for i, item in enumerate(out_data):
+                print(f"Dictionary {i}:")
+                if isinstance(item, dict):
+                    for key, value in item.items():
+                        print(f"  Key: '{key}' has value '{value}' of type {type(value)}")
+                else:
+                    print(f"Item {i} is not a dictionary.")
+
+        check_dict_types(data)
+        
+        # in id: 0, returns c3 : 16.0 && in id: 1, returns c3: nan
+        
+        print(out_data)
+        
+    
+    def test_avg0_distinct(self):
+        sql = SQLContext('test_avg0_distinct', TEST_CLIENT).get_or_create()
+        
+        sql.register_table_from_sql(''' 
+                                    CREATE TABLE avg0_distinct(
+                                        c1 TINYINT,
+                                        c2 TINYINT NOT NULL,
+                                        c3 INT2,
+                                        c4 INT2 NOT NULL,
+                                        c5 INT,
+                                        c6 INT NOT NULL,
+                                        c7 BIGINT,
+                                        c8 BIGINT NOT NULL);
+                                    ''')
+
+
+        sql.register_materialized_view("avg_view_distinct", '''
+                                       SELECT
+                                            AVG(DISTINCT c1) AS c1,
+                                            AVG(DISTINCT c2) AS c2,
+                                            AVG(DISTINCT c3) AS c3,
+                                            AVG(DISTINCT c4) AS c4,
+                                            AVG(DISTINCT c5) AS c5,
+                                            AVG(DISTINCT c6) AS c6,
+                                            AVG(DISTINCT c7) AS c7,
+                                            AVG(DISTINCT c8) AS c8
+                                       FROM avg0_distinct''')
+        data = [
+            {"c1": 1, "c2": 2, "c3": 3, "c4": 4, "c5": 5, "c6": 6, "c7": None, "c8": 8},
+            {"c1": None, "c2": 2, "c3": 30, "c4": 4, "c5": None, "c6": 60, "c7": 70, "c8": 8},
+            {"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        
+        out = sql.listen("avg_view_distinct")
+        
+        sql.start()
```

Comment:
you like empty lines a lot. I would use empty lines only infrequently. space on the screen is an important resource